### PR TITLE
Feature: Comments

### PR DESCRIPTION
**Pull request by henderkes**
…usermanagement is missing, will be done soon

### CODE REVIEW

File: backend/api/projects.py [hunk 1]
```diff
@@ -187,3 +187,190 @@ def update_project(project_id):
...
+        response: Success response
+                  or 404 if project is not found
+                  or 400 if validation error occurs
+    """
+    try:
+        manifest = g.projects.find_one({'_id': project_id})
+        if manifest is None:
+            raise ApiException("Project not found", 404)
+        if "text/plain" not in request.content_type:
+            raise ApiException("Comment had not format string", 400)
```
Comment (Drakulix) [inline backend/api/projects.py]: Wording seems to be a bit off here.

File: backend/api/projects.py [hunk 2]
```diff
@@ -187,3 +187,190 @@ def update_project(project_id):
...
+        comment = {}
+        comment[author] = None  # How to get this
```
Comment (Drakulix) [inline backend/api/projects.py]: `from flask_security import current_user`

**Review by Drakulix — COMMENTED** [review]
Already very good.

File: backend/api/projects.py (hunk 1 above)
Comment (henderkes) [reply backend/api/projects.py]: yes I'm already working on it and tweaking a few things + implementing current user

File: backend/manifest_schema.json [hunk 3]
```diff
@@ -152,6 +152,21 @@
...
+        "type": "array",
+        "items": {
+          "author": {
+            "$ref": "#/project/definitions/user"
+          },
+          "commentId": {
+            "type": "string",
+            "format": "uuid"
+          },
+          "commentMsg": {
```
Comment (henderkes) [inline backend/manifest_schema.json]: shouldnt this be "comment_id" and "comment_msg" to follow our naming?

Comment (Drakulix) [reply backend/manifest_schema.json]: You are right, I would even prefer this to be named `id` and `message`.

**Comment by henderkes** [comment]
woops, defined the method before the User class

**Comment by henderkes** [comment]
Do we maybe have to define one field of the user as primary=True?

**Comment by henderkes** [comment]
And apparentpy I forgot pushing my last changes, email should be unique, too.

File: backend/api/projects.py [hunk 4]
```diff
@@ -248,11 +248,11 @@ def add_comment(project_id):
...
     """
     try:
-        manifest = g.projects.find({'_id': project_id}, {'comments': 1, '_id': 0}).sort(
+        comments = g.projects.find({'_id': project_id}, {'comments': 1, '_id': 0}).sort(
                    key=lambda x: x['datetime'], reverse=True)
-        if manifest is None:
+        if comments is None:
             raise ApiException("Project not found", 404)
-        return jsonify(manifest['comments'])
+        return jsonify(comments)
```
Comment (henderkes) [inline backend/api/projects.py]: I dont think this is the intended return. The (list of) comments should be returned, not the entire json I think?

**Comment by Drakulix** [comment]
@DubbleClick Is this now ready for merge except for this one nitpick?

**Pull request merged**